### PR TITLE
Fix duplicate path in old Network API versions (Network January branch)

### PR DESCRIPTION
**Pull request by anton-evseev**
Pick https://github.com/Azure/azure-rest-api-specs/pull/4940 and https://github.com/Azure/azure-rest-api-specs/pull/5002 from `master` into January branch

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by number213** [comment]
Travis fails with OOM exception
```
v8::Utils::ReportOOMFailure(char const*, bool) [node]\n 4: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool)
```
I tried linter on my local machine and had the same error. Managed to bypass it by using `--max_old_space_size=8192`.

@dsgouda @jianghaolu could you please take a look?

File: specification/network/resource-manager/Microsoft.Network/stable/2015-06-15/virtualNetworkGateway.json [hunk 1]
```diff
@@ -437,45 +437,6 @@
         "x-ms-long-running-operation": true
       }
     },
-    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/connections/{connectionSharedKeyName}/sharedkey": {
```
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/stable/2015-06-15/virtualNetworkGateway.json]: @number213  can you explain how you are fixing duplicate nw paths here

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks fine apart from the comment.

**Comment by number213** [comment]
CI failures are related to this issue https://github.com/Azure/azure-rest-api-specs/pull/5008

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**